src/04-modality-specific-files/03-electroencephalography.md

```diff
@@ -275,6 +279,8 @@ UADC001  MISC   n/a     envelope of audio signal        good    n/a
 
 Template:
 
+{{ make_filename_template(datatypes=["eeg"], suffixes=["electrodes"]) }}
```

Comment:
this produces a template that includes a `electrodes.json`, however there is no such file. Metadata for `electrodes.tsv` is specified in `coordsystem.json`:

![image](https://user-images.githubusercontent.com/9084751/97555781-d8735700-19d8-11eb-9b62-f6202e58a6f6.png)

Reply:
also, I think that the `ses-label` would need to go on a new line (and indentation appropriately fixed)

Lastly, we should have a discussion on whether we do want the square brackets to indicate optional Req. Level. or not --> and then apply that consistently. It's currently not documented anywhere. (This should be a new issue)

Reply:
re `[]` -- aren't they are used already throughout bids specification (see e.g. [anatomical](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#anatomy-imaging-data))? if anything we indeed just need clarification somewhere in "common principles", but IMHO it isn't specific to this PR anyhow

Reply:
re `_electrodes.json` ... eh indeed, there is no `_electrodes.json` but `_electrodes.tsv` must be accompanied with `_coordsystem.json` [apparently](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#electrodes-description-_electrodestsv).  Not sure if that is anyhow expressible in the schema ATM.
Also I noted in there a use of `[]` to annotate that an entire file is "optional" (`[sub-<label>[_ses-<label>][_acq-<label>]_coordsystem.json]`) -- not sure if we should do that.  Majority of the files are optional or with some tricky dependencies like here `_electrodes.tsv` requires `_coordsystem.json`

Reply:
> eh indeed, there is no _electrodes.json but _electrodes.tsv must be accompanied with _coordsystem.json apparently.

yes. 

> Also I noted in there a use of [] to annotate that an entire file is "optional"

yup, that's what I meant. I think we should stop doing that, and document [] for option ENTITIES in a filename.

(in a separate PR)

Reply:
> also, I think that the ses-label would need to go on a new line (and indentation appropriately fixed)

This is done inconsistently in the specification. In MEG, iEEG, and EEG, the templates have a mix of two-space and four-space tabs and `ses` is on a separate line from `sub`, but in the rest of the sections, they use four-space tabs and `ses` is on the same line as `sub`. I went with the latter, but can change if you prefer the former.

>eh indeed, there is no _electrodes.json but _electrodes.tsv must be accompanied with _coordsystem.json apparently. Not sure if that is anyhow expressible in the schema ATM.

I have an open issue about that kind of logic (#620), but no actual solution just yet. The same goes for labeling entire files as optional. In the mean time, I can just remove `json` as a valid extension for `electrodes` files.

Reply:
Actually I think fixing `electrodes.json` should probably go in a separate PR since it's an issue with the actual schema. I'll open one ASAP.

Reply:
let's continue discussion in the main thread: https://github.com/bids-standard/bids-specification/pull/610#issuecomment-718852640 ... this code-review-comment thread can easily become "invisible"